lib/executeHook.js

```diff
@@ -83,12 +86,12 @@ module.exports = function(server, settings, collection, hook) {
             } else {
               updateHook.results.push({ action, output });
             }
-            return eachDone();
+            return finish();
```

Comment:
Why do we need try catch? Seems like that just muddies up all the call back logic